Sandbox/RPG/Player.h

```diff
@@ -177,6 +200,95 @@ class Player
 		m_playerPreviousState = m_playerState;
 	}
 
+	void addItem(Item* item)
+	{
+		// If the bucket for this item type exists
+		if (m_inventory.find(item->m_name) == m_inventory.end()) 
+		{
+			m_inventory[item->m_name] = std::vector<Item*>{ item };
+			m_itemTypes.push_back(item->m_name);
+			m_activeItemIt = m_itemTypes.end() - 1;
+		}
+		else 
+		{
+			m_inventory[item->m_name].push_back(item);
+		}
+	}
+
+	int getItemCount(const std::string& item)
+	{
+		if (m_inventory.find(item) == m_inventory.end()) 
+		{
+			return 0;
+		}	
+		else 
+		{
+			return m_inventory[item].size();
+		}
+	}
+
+	Item* popItem(const std::string& item)
+	{
+		if (m_inventory.find(item) == m_inventory.end())
+		{
+			return nullptr;
+		}
+		else 
+		{
+			Item* lastItem = m_inventory[item].back();
+			m_inventory[item].pop_back();
+
+			// All items are gone 
+			if (m_inventory[item].size() == 0)
+			{
+				m_inventory.erase(item);
+
+				if (item == *m_activeItemIt)
+				{
+					m_activeItemIt = m_itemTypes.erase(m_activeItemIt);
+					if (m_activeItemIt == m_itemTypes.end())
+					{
+						m_activeItemIt = m_itemTypes.begin();
+					}
+				}
+			}
```

Comment:
If this runs when an item of a certain type runs out, do you think it would be more performant to maintain the empty vector, and just check if the particular inventory vector is empty when iterating through the whole inventory to skip over items that have an empty vector?

This is just me thinking out loud about performance, mainly I am just curious if it more performant to maintain empty item vectors and skip over them upon iteration, as opposed to destructing when empty, and then constructing again if an item of that type is picked up in the hypothetical RPG for example